### PR TITLE
Move rake test:* tasks to all use `bin/rspec`.

### PR DESCRIPTION
**Pull request by andrewvc**
This eliminates the differences in behavior between bin/rspec and rake test:*

### CODE REVIEW

File: rakelib/test.rake [hunk 1]
```diff
@@ -6,66 +6,35 @@ require 'pathname'
...
+  task "core-java" do
     exit(1) unless system './gradlew clean test --info'
-    
-    specs = ["spec/unit/**/*_spec.rb", "logstash-core/spec/**/*_spec.rb"]
-
-    Rake::FileList[*specs]
   end
 
   desc "run all core specs"
   task "core" => ["core-slow"]
```
Comment (original-brownbear) [inline rakelib/test.rake]: @andrewvc This one was what ran the Java tests also. We must should `core-java` as a depend task for it right?

Comment (andrewvc) [inline rakelib/test.rake]: Yeah, great point, I'll add that!

**Comment by andrewvc** [comment]
@original-brownbear the original version of these tasks ran the java tests for all core tests. I would argue that we should skip the java tests on `core-fast`. If you really care about them you can run `core:java` IMHO. WDYT?

/cc @suyograo

**Comment by original-brownbear** [comment]
@andrewvc agreed, the `core` ones are also what's run on Travis/CI and that's what matters imo, as long as we don't lose our default CI `test` task running the Java tests we're good I think :)

**Comment by original-brownbear** [comment]
@andrewvc in that sense, LGTM :)

**Pull request closed without merging**